src/proxy/MultiMapProxy.ts

```diff
@@ -0,0 +1,204 @@
+import {MultiMapKeySetCodec} from './../codec/MultiMapKeySetCodec';
+import {MultiMapValuesCodec} from './../codec/MultiMapValuesCodec';
+import {MultiMapEntrySetCodec} from './../codec/MultiMapEntrySetCodec';
+import {MultiMapContainsKeyCodec} from './../codec/MultiMapContainsKeyCodec';
+import {MultiMapContainsValueCodec} from './../codec/MultiMapContainsValueCodec';
+import {MultiMapContainsEntryCodec} from './../codec/MultiMapContainsEntryCodec';
+import {MultiMapSizeCodec} from './../codec/MultiMapSizeCodec';
+import {MultiMapClearCodec} from './../codec/MultiMapClearCodec';
+import {MultiMapValueCountCodec} from './../codec/MultiMapValueCountCodec';
+import {EntryEventType} from '../core/EntryEventType';
+import ClientMessage = require('../ClientMessage');
+import {MultiMapAddEntryListenerToKeyCodec} from './../codec/MultiMapAddEntryListenerToKeyCodec';
+import {MultiMapAddEntryListenerCodec} from './../codec/MultiMapAddEntryListenerCodec';
+import {MultiMapRemoveEntryListenerCodec} from './../codec/MultiMapRemoveEntryListenerCodec';
+import {MultiMapLockCodec} from '../codec/MultiMapLockCodec';
+import {MultiMapIsLockedCodec} from '../codec/MultiMapIsLockedCodec';
+import {MultiMapTryLockCodec} from '../codec/MultiMapTryLockCodec';
+import {MultiMapUnlockCodec} from '../codec/MultiMapUnlockCodec';
+import {MultiMapForceUnlockCodec} from '../codec/MultiMapForceUnlockCodec';
+
+export class MultiMapProxy<K, V> extends BaseProxy implements MultiMap<K, V> {
+
+    put(key: K, value: V): Promise<boolean> {
+        var keyData = this.toData(key);
+        var valueData = this.toData(value);
+        return this.encodeInvokeOnKey<boolean>(MultiMapPutCodec, keyData, keyData, valueData, 1);
+    }
+
+    get(key: K): Promise<Array<V>> {
+        var keyData = this.toData(key);
+
+        var deserializeCallback = (items: Array<Data>) => {
+            return items.map((item) => {
+                return this.toObject(item);
+            });
+        };
+
+        return this.encodeInvokeOnKey<Array<Data>>(MultiMapGetCodec, keyData, keyData, 1)
+            .then<Array<V>>(deserializeCallback);
+    }
+
+    remove(key: K, value: V): Promise<boolean> {
+        var keyData = this.toData(key);
+        var valueData = this.toData(value);
+        return this.encodeInvokeOnKey<boolean>(MultiMapRemoveEntryCodec, keyData, keyData, valueData, 1);
+    }
+
+    removeAll(key: K): Promise<Array<V>> {
+        var keyData = this.toData(key);
+        return this.encodeInvokeOnKey<Array<Data>>(MultiMapRemoveCodec, keyData, keyData, 1)
+            .then<Array<V>>((items: Array<Data>) => {
+                return items.map((item) => {
+                    return this.toObject(item);
+                });
+            });
+    }
+
+    keySet(): Promise<Array<K>> {
+        return this.encodeInvokeOnRandomTarget<Array<Data>>(MultiMapKeySetCodec)
+            .then<Array<K>>((items: Array<Data>) => {
```

Comment:
Actually I see this piece of code is used everywhere in this method. Why not just extract this as a private method and use it everywhere rather than creating another function everytime?